status-server/src/public/js/client.js

```diff
@@ -1,199 +1,168 @@
+    if (err) {
+      console.error('Could not request my status');
+      return;
+    }
 
-function stopPolling() {
-  if (intervalID) {
-    clearInterval(intervalID);
-  }
-  intervalID = null;
-}
+    appState.mySlot.status = data;
 
-function togglePolling(btn) {
-  if (intervalID) {
-    stopPolling();
-    if(btn) {
-      btn.value = "Start";
-    }
-  } else {
-    startPolling();
-    if (btn) {
-      btn.value = "Stop";
-    }
-  }
+    renderApp();
+  });
+
+  getMySlots();
+
+  var intervalID = setInterval(getMySlots, 2000);
 }
 
-function init() {
-  // update the main heading if this client instance was passed an id
-  if (config) {
-    // assign a default id if none was passed
-    if (!config.id) {
-      config.id = 'user0';
-    }
-    var titleNode = document.querySelector('h1');
-    if (titleNode) {
-      titleNode.innerHTML = 'status client: ' + config.id;
+function getMySlots() {
+  jsonRequest('GET', 'user/'+ config.id +'/slots', function(err, data) {
+    if (err) {
+      console.error('Could not request my slots');
+      return;
     }
-    // temporarily map the user id to the led node just by aligning the indices.
-    // We'll want a proper slots=>users mapping eventually
-    Array.prototype.forEach.call(document.querySelectorAll('.led'), (node, idx) => {
-      var userId = node.id.replace(/^led([0-9])+/, 'user$1');
-      node.dataset.user = userId;
-      if (config && config.id.endsWith(idx)) {
-        node.classList.add('isSelf');
+
+    appState.slots = data.value.map(function(id){
+      if (id === null) {
+        return null;
       }
+      
+      return {
+        id: id,
+        status: null
+      };
     });
-  }
 
-  if (config && config.autostart) {
-    togglePolling(document.getElementById('pollingBtn'));
-  }
+    var reqTotal = 0;
+    var reqComplete = 0;
+
+    appState.slots.forEach(function(slot) {
+      if (slot === null) {
+        return;
+      }
+
+      reqTotal += 1;
+
+      jsonRequest('GET', 'user/'+ slot.id +'/status', function(err, data) {
+        if (err) {
+          console.error('Could not request status of' + slot.id);
+          return;
+        }
+        slot.status = data;
+        
+        reqComplete += 1;
+        if (reqComplete === reqTotal) {
+          renderApp();
+        }
+      });
+    });
+  });
 }
-function handleLEDClick(ct) {
-  var urlKey = '/user/'+ct+'/status';
-  // NOTE: we'll actually need to match user id here once we have a proper way to represent users
-  // for now I'm just checking the last character is the same digit
-  var isSelf = config && config.id.endsWith(ct);
-  console.log('isSelf: ', isSelf);
-  // NOTE: for now, click just means update status, and we can only update our own status
-  // so ignore clicks on other LEDs
-  if (isSelf) {
-    postNewStatus(urlKey);
-  } else {
-    console.log('Ignoring click on LED for '+ct+' that isnt the status of me: ' + config.id);
+
+function jsonRequest(method, path, data, callback) {
```

Comment:
I can go through the jsonRequest functions and convert them to promises. It will take me a little time as I am not too comfortable with them yet, but we can see which version of the code is less lines.